django/db/models/sql/query.py

```diff
@@ -1996,10 +1996,11 @@ def split_exclude(self, filter_expr, can_reuse, names_with_path):
             lookup = lookup_class(pk.get_col(query.select[0].alias), pk.get_col(alias))
             query.where.add(lookup, AND)
             query.external_aliases[alias] = True
+        else:
+            lookup_class = select_field.get_lookup("exact")
+            lookup = lookup_class(col, ResolvedOuterRef(trimmed_prefix))
+            query.where.add(lookup, AND)
```

Comment:
When I initially stepped into this function wrt this ticket I was also thinking of excluding this particular block so good to see my assumption confirmed 👍